detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/parse/GradleReportParser.java

```diff
@@ -38,24 +41,27 @@ public Optional<GradleReport> parseReport(File reportFile) {
         List<String> configurationLines = new ArrayList<>();
         try (InputStream dependenciesInputStream = new FileInputStream(reportFile);
             BufferedReader reader = new BufferedReader(new InputStreamReader(dependenciesInputStream, StandardCharsets.UTF_8))) {
-            while (reader.ready()) {
-                String line = reader.readLine();
-                /*
-                  The meta data section will be at the end of the file after all of the "gradle dependencies" output
-                 */
-                if (line.startsWith(DETECT_META_DATA_HEADER)) {
+
+            List<String> reportLines = reader.lines().collect(Collectors.toList());
+
+            for(int i = reportLines.size()-1; i>=0; i--) {
+                if (reportLines.get(i).startsWith(DETECT_META_DATA_FOOTER)) {
                     processingMetaData = true;
-                } else if (line.startsWith(DETECT_META_DATA_FOOTER)) {
-                    processingMetaData = false;
+                } else if (reportLines.get(i).startsWith(DETECT_META_DATA_HEADER)) {
+                    break;
                 } else if (processingMetaData) {
-                    setGradleReportInfo(gradleReport, line);
+                    setGradleReportInfo(gradleReport, reportLines.get(i));
+                }
+            }
+
+            for(String line: reportLines) {
+                if (StringUtils.isBlank(line)) {
+                    parseConfigurationLines(configurationLines, gradleReport);
+                    configurationLines.clear();
+                } else if (line.startsWith(DETECT_META_DATA_HEADER)) {
+                    break;
                 } else {
-                    if (StringUtils.isBlank(line)) {
-                        parseConfigurationLines(configurationLines, gradleReport);
-                        configurationLines.clear();
-                    } else {
-                        configurationLines.add(line);
-                    }
+                    configurationLines.add(line);
```

Comment:
In all the dependency graph files which are generated, the metadata information is at the end of the file, so we start reading the file from the end and once we have read the metadata, we start reading the file normally.